crates/sui-graphql-rpc/src/types/dynamic_field.rs

```diff
@@ -109,7 +104,10 @@ impl DynamicField {
             let obj = MoveObject::query(
                 ctx.data_unchecked(),
                 self.df_object_id,
-                ObjectLookupKey::Latest,
+                match self.super_.super_.checkpoint_viewed_at {
```

Comment:
This needs to use a version-bounded query, not a checkpoint-bounded query.

Reply:
This PR comment was marked as resolved, but the query still seems to be checkpoint bounded, not version bounded -- was this fixed up the stack?

Reply:
No, I did not understand what you were saying until now. But I think I've caught on, and I've added some tests around this